### PR TITLE
Update .travis.yml to test on macOS

### PR DESCRIPTION
**Pull request by talentdeficit**

### CODE REVIEW

**Comment by josevalim** [comment]
Thank you. This will jump to latest Erlang in the future and we may not be ready. Plus, once we branch, this can get outdated. Is there any way to pick a version? Maybe we should use kerl instead?

**Comment by talentdeficit** [comment]
it's possible to pin a particular version (via an external tap) but i thought it was better to test based on the [install instructions](http://elixir-lang.org/install.html#mac-os-x) so that breaking changes are detected asap

**Comment by josevalim** [comment]
@talentdeficit Brew installing the latest is possibly OK for development but it is not a good one for CIs. Going to any branch 2 years later will likely mean a failing build. The install instructions are always against the latest.

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,29 @@
...
+    - os: linux
+      otp_release: 19.0
+    - os: linux
+      otp_release: 19.1
+    - os: linux
+      otp_release: 19.2
+    - os: osx
+      language: generic
 
-sudo: false
```
Comment (kelvinst) [inline .travis.yml]: Is there a reason for removing the `sudo: false` configuration?

Comment (talentdeficit) [inline .travis.yml]: `sudo: false` is the default

Comment (kelvinst) [inline .travis.yml]: Great! Sorry for bothering you about it. 😬

Comment (talentdeficit) [inline .travis.yml]: actually, it turns out travis uses the date the repo was added to travis to determine the default, so `sudo: false` is required. i'll fix it

**Comment by edgurgel** [comment]
I may be completely wrong but I think you can do `brew install https://...` which points to a specific file from a specific commit hash which could be used to install Erlang 19.2 for example.

**Comment by talentdeficit** [comment]
i've updated this to include osx but to not fail the build if the osx build fails. that makes it advisory. is that more acceptable @josevalim ?

**Comment by talentdeficit** [comment]
@edgurgel brew supports versions in formulas now, but no one has yet done the work to update the erlang formula. i plan to do it but i don't have time today. once that is done it will be possible to test only against a specific version

**Comment by lexmag** [comment]
I'm not sure if we should add testing on macOS considering early support for it on Travis, and possible build failures or timeouts. 🤔

**Comment by josevalim** [comment]
@lexmag with allowed failures it makes that less of an issue. We can give it a try without affecting our CI status. :)

**Comment by fishcakez** [comment]
Thank you :heart:

**Pull request merged**